backend/hdf5/DimensionHDF5.cpp

```diff
@@ -259,13 +269,156 @@ void SetDimensionHDF5::labels(const none_t t) {
+    this->group.createLink(target->group(), "data_frame");
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame,
+                                               unsigned col_index)
+    :DataFrameDimensionHDF5(group, index, file, block, frame)
+{
+    this->group.setAttr("column_index", col_index);
+}
+
+boost::optional<unsigned> DataFrameDimensionHDF5::columnIndex() const {
+    unsigned idx;
+    boost::optional<unsigned> col_index;
+
+    if (group.hasAttr("column_index")) {
+        group.getAttr("column_index", idx);
+        col_index = idx;
+    }
+
+    return col_index;
+}
+
+boost::optional<unsigned> DataFrameDimensionHDF5::checkColumnIndex(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index = col_index;
+    if (!col_index) {
+        column_index = columnIndex();
+    }
+
+    if (!column_index) {
+        throw nix::OutOfBounds("DataFrameDimensionHDF5: Error accessing column, no column index was given and no default is specified.");
+    }
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
+    if (static_cast<size_t>(*column_index) >= cols.size()) {
+        throw nix::OutOfBounds("DataFrameDimensionHDF5: Error accessing column, column index exceeds number of columns!");
+    }
+
+    return column_index;
+}
+
+
+DimensionType DataFrameDimensionHDF5::dimensionType() const {
+    return DimensionType::DataFrame;
+}
+
+
+std::string DataFrameDimensionHDF5::unit(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index = checkColumnIndex(col_index);
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
+    return cols[*column_index].unit;
+}
+
+
+std::string DataFrameDimensionHDF5::label(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index;
+    nix::DataFrame df = dataFrame();
+    if (!col_index) {
+        column_index = columnIndex();
+        if (!column_index) {
+            return df.name();
+        }
+    } else {
+        column_index = col_index;
+    }
+    column_index = checkColumnIndex(column_index);
+
+    std::string name = df.colName(*column_index);
+    return name;
+}
+
+
+Column DataFrameDimensionHDF5::column(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index = checkColumnIndex(col_index);
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
+    return cols[*column_index];
+}
+
+
+nix::DataType DataFrameDimensionHDF5::columnDataType(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index = checkColumnIndex(col_index);
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> all_cols = df.columns();
```

Comment:
Also lame that we have to fetch all of them if we only want one. Filed as issue #809